app/src/sharedTest/java/org/oppia/app/home/HomeActivityTest.kt

```diff
@@ -89,10 +99,185 @@ class HomeActivityTest {
+    launch(HomeActivity::class.java).use {
+      onView(
+        atPositionOnView(
+          R.id.home_recycler_view,
+          0,
+          R.id.welcome_text_view
+        )
+      ).check(matches(withText(containsString("Welcome"))))
+    }
+  }
+
+  @Test
+  fun testHomeActivity_recyclerViewIndex0_configurationChange_displaysWelcomeMessageCorrectly() {
+    launch(HomeActivity::class.java).use {
+      it.onActivity { activity ->
+        activity.requestedOrientation = Configuration.ORIENTATION_LANDSCAPE
+      }
+      onView(
+        atPositionOnView(
+          R.id.home_recycler_view,
+          0,
+          R.id.welcome_text_view
+        )
+      ).check(matches(withText(containsString("Welcome"))))
+    }
+  }
+
+  @Test
+  fun testHomeActivity_recyclerViewIndex1_displaysRecentlyPlayedStoriesText() {
+    launch(HomeActivity::class.java).use {
+      onView(atPositionOnView(R.id.home_recycler_view, 1, R.id.recently_played_stories_text_view)).check(
+        matches(
+          withText(R.string.recently_played_stories)
+        )
+      )
+    }
+  }
+
+  @Test
+  fun testHomeActivity_recyclerViewIndex1_displaysViewAllText() {
+    launch(HomeActivity::class.java).use {
+      onView(atPositionOnView(R.id.home_recycler_view, 1, R.id.view_all)).check(
+        matches(
+          withText(R.string.view_all)
+        )
+      )
+    }
+  }
+
+  @Test
+  fun testHomeActivity_recyclerViewIndex1_clickViewAll_opensContinuePlayingActivity() {
+    launch(HomeActivity::class.java).use {
+      onView(atPositionOnView(R.id.home_recycler_view, 1, R.id.view_all)).perform(click())
+      intended(hasComponent(ContinuePlayingActivity::class.java.name))
+    }
+  }
+
+  @Test
+  fun testHomeActivity_recyclerViewIndex1_promotedCard_chapterNameIsCorrect() {
+    launch(HomeActivity::class.java).use {
+      onView(atPositionOnView(R.id.home_recycler_view, 1, R.id.chapter_name)).check(
+        matches(
+          withText(containsString("The Meaning of Equal Parts"))
+        )
+      )
+    }
+  }
+
+  @Test
+  fun testHomeActivity_recyclerViewIndex1_promotedCard_storyNameIsCorrect() {
+    launch(HomeActivity::class.java).use {
+      onView(atPositionOnView(R.id.home_recycler_view, 1, R.id.story_name)).check(
+        matches(
+          withText(containsString("Second Story"))
+        )
+      )
+    }
+  }
+
+  @Test
+  fun testHomeActivity_recyclerViewIndex1_configurationChange_promotedCard_storyNameIsCorrect() {
+    launch(HomeActivity::class.java).use {
+      it.onActivity { activity ->
+        activity.requestedOrientation = Configuration.ORIENTATION_LANDSCAPE
+      }
+      onView(atPositionOnView(R.id.home_recycler_view, 1, R.id.story_name)).check(
+        matches(
+          withText(containsString("Second Story"))
+        )
+      )
+    }
+  }
+
+  // TODO(#135): Update this test case to check that topic-play tab is opened.
+  @Test
+  fun testHomeActivity_recyclerViewIndex1_clickPromotedStory_opensTopicActivity() {
```

Comment:
Is it worth verifying that we are also sending the story ID? See my above comment--we should verify we're using the correct API here.

Reply:
Done. Added verification for topicId and storyId.